cargo-shuttle/src/config.rs

```diff
@@ -249,10 +262,75 @@ impl RequestContext {
         Ok(Self {
             global,
             project: None,
+            project_internal: None,
             api_url: None,
         })
     }
 
+    pub fn load_local_internal(&mut self, project_args: &ProjectArgs) -> Result<()> {
+        let workspace_path = project_args
+            .workspace_path()
+            .unwrap_or(project_args.working_directory.clone());
+
+        trace!(
+            "looking for .shuttle/config.toml in {}",
+            workspace_path.display()
+        );
+        let local_manager =
+            LocalConfigManager::new(workspace_path, ".shuttle/config.toml".to_string());
+        let mut project_internal = Config::new(local_manager);
+        if !project_internal.exists() {
+            trace!("no local .shuttle/config.toml found");
+            project_internal.replace(InternalProjectConfig::default());
+        } else {
+            trace!("found a local .shuttle/config.toml");
+            project_internal.open()?;
+        }
+
+        let config = project_internal.as_mut().unwrap();
+
+        // Project id is preferred in this order:
+        // 1. Name given on command line
+        // 2. Name from .shuttle/config.toml file
+        match (&project_args.name, &config.id) {
+            // Command-line name parameter trumps everything
+            (Some(id_from_args), _) => {
+                trace!("using command-line project id");
+                config.id = Some(id_from_args.clone());
+            }
+            // If key exists in config then keep it as it is
+            (None, Some(_)) => {
+                trace!("using .shuttle/config.toml project id");
+            }
+            (None, None) => {
+                trace!("no project id in args or config found");
+            }
+        };
```

Comment:
It's a bit confusing that this is name or id, but we expect it to be id.

Reply:
If a name is given, it is translated to its id on the go in `load_project`